dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/InstallKernelMap.java

```diff
@@ -708,4 +764,79 @@ private Integer uninstall() {
         }
         return OK;
     }
-}
+
+    private static void populateShortNameFromManifest(File esa, Map<String, String> shortNameMap) throws IOException {
+        String esaLocation = esa.getCanonicalPath();
+        ZipFile zip = null;
+        try {
+            zip = new ZipFile(esaLocation);
+            Enumeration<? extends ZipEntry> zipEntries = zip.entries();
+            ZipEntry subsystemEntry = null;
+            while (zipEntries.hasMoreElements()) {
+                ZipEntry nextEntry = zipEntries.nextElement();
+                if ("OSGI-INF/SUBSYSTEM.MF".equalsIgnoreCase(nextEntry.getName())) {
+                    subsystemEntry = nextEntry;
+                    break;
+                }
+            }
+            if (subsystemEntry != null) {
+                Manifest m = ManifestProcessor.parseManifest(zip.getInputStream(subsystemEntry));
+                Attributes manifestAttrs = m.getMainAttributes();
+                String shortNameAttr = manifestAttrs.getValue(SHORTNAME_HEADER_NAME);
+                shortNameMap.put(esa.getCanonicalPath(), shortNameAttr);
+            }
+        } finally {
+            if (zip != null) {
+                zip.close();
+            }
+        }
+    }
+
+    /**
+     * generate a JSON from provided individual ESA files
+     *
+     * @param generatedJson path
+     * @param shortNameMap contains features parsed from individual esa files
+     * @return singleJson file
+     * @throws IOException
+     * @throws BuildException
+     * @throws RepositoryException
+     * @throws InstallException
+     */
+    private File generateJsonFromIndividualESAs(Path jsonDirectory, Map<String, String> shortNameMap) throws IOException, BuildException, RepositoryException, InstallException {
+
+        String dir = jsonDirectory.toString();
+        List<File> esas = (List<File>) data.get(INDIVIDUAL_ESAS);
+        File singleJson = new File(dir + "/SingleJson.json");
+
+        for (File esa : esas) {
+            try {
+                populateShortNameFromManifest(esa, shortNameMap);
+            } catch (IOException e) {
+                throw new InstallException(Messages.INSTALL_KERNEL_MESSAGES.getLogMessage("ERROR_ESA_NOT_FOUND", esa.getCanonicalPath()));
```

Comment:
Use `esa.getAbsolutePath()` for the message, otherwise getting canonical path could throw another exception which is not caught.